app/src/lib/stores/app-store.ts

```diff
@@ -2512,6 +2530,46 @@ export class AppStore extends TypedBaseStore<IAppState> {
       return repository
     }
 
+    const { changesState, branchesState } = this.repositoryStateCache.get(
+      repository
+    )
+    const { tip } = branchesState
+    const currentBranch = tip.kind === TipState.Valid ? tip.branch : null
+    const hasChanges = changesState.workingDirectory.files.length > 0
+
+    if (
+      enableStashing() &&
+      hasChanges &&
+      currentBranch !== null &&
+      uncommittedChangesStrategy ===
+        UncommittedChangesStrategy.askForConfirmation
+    ) {
+      this._showPopup({
+        type: PopupType.StashAndSwitchBranch,
+        branchToCheckout: foundBranch,
+        repository,
+      })
+
+      return repository
+    }
+
+    let shouldPopStash = false
+
+    if (currentBranch !== null) {
```

Comment:
Put behind feature flag

Reply:
Resolved in e8aad48504410a5954c524f2b443a77a707f1eda